lib/charms/vault_k8s/v0/juju_facade.py

```diff
@@ -315,6 +337,30 @@ def set_secret_expiry(self, expiry: datetime, label: str, id: str | None = None)
         except TransientJujuError:
             raise
 
+    def grant_secret_to_relation(
+        self,
+        relation: Relation,
+        secret: Secret | None = None,
+        secret_id: int | None = None,
+        secret_label: str | None = None,
+    ) -> None:
+        """Grant the secret to the relation.
+
+        If secret ID or label is provided, it retrieves the secret and grants it to the relation.
+
+        Raises:
+            NoSuchSecretError
+            SecretRemovedError
+        """
+        if secret:
+            secret.grant(relation)
+            return
+        try:
+            secret = self.get_secret(id=secret_id, label=secret_label)
+        except (NoSuchSecretError, SecretRemovedError):
+            raise
+        secret.grant(relation)
```

Comment:
The secret is not granted to the relation, but to the app and units on the other side of the relation. The The relation is used to scope the life of the secret.

Reply:
Renamed.